test/groovy/com/sap/piper/cm/StepHelpersTest.groovy

```diff
@@ -209,4 +210,176 @@ class StepHelpersTest extends BasePiperTest {
+    }
+
+    @Test
+    public void reqidNotProvidedTest() {
+
+        def calledWithParameters = null
+    
+        helper.registerAllowedMethod( 'piperExecuteBin', [Map, String, String, List],
+            {
+                params, stepName, metaData, creds -> 
+                    if(stepName.equals("transportRequestReqIDFromGit")) {
+                        calledWithParameters = params
+                    }
+            }
+        )
+
+        loggingRule.expect('[INFO] Retrieving transportRequestId from commit history')
+        loggingRule.expect('[WARN] Cannot retrieve transportRequestId from commit history')
+
+        def transportRequestId = StepHelpers.getTransportRequestId(nullScript, params)
+
+        assert transportRequestId == null
+        assert calledWithParameters != null
+    }
+
+    @Test
+    public void docidViaConfigTest() {
+
+        nullScript.commonPipelineEnvironment.setValue('changeDocumentId', 'unused')
+        
+        def calledWithParameters = null
+    
+        helper.registerAllowedMethod( 'piperExecuteBin', [Map, String, String, List],
+            {
+                params, stepName, metaData, creds ->
+                    if(stepName.equals("transportRequestDocIDFromGit")) {
+                        calledWithParameters = params
+                    }
+            }
+        )
+
+        def changeDocumentId = StepHelpers.getChangeDocumentId(nullScript, [changeDocumentId: '100'])
+
+        assert changeDocumentId == '100'
+        assert calledWithParameters == null
+    }
+
+    @Test
+    public void docidViaCommonPipelineEnvironmentTest() {
+
+        nullScript.commonPipelineEnvironment.setValue('changeDocumentId', '200')
+        
+        def calledWithParameters = null
+    
+        helper.registerAllowedMethod( 'piperExecuteBin', [Map, String, String, List],
+            {
+                params, stepName, metaData, creds -> 
+                    if(stepName.equals("transportRequestDocIDFromGit")) {
+                        calledWithParameters = params
+                    }
+            }
+        )
+
+        def changeDocumentId = StepHelpers.getChangeDocumentId(nullScript, params)
+
+        assert changeDocumentId == '200'
+        assert calledWithParameters == null
+    }
+
+    @Test
+    public void docidViaCommitHistoryTest() {
+
+        def calledWithParameters = null
+    
+        helper.registerAllowedMethod( 'piperExecuteBin', [Map, String, String, List],
+            {
+                params, stepName, metaData, creds -> 
+                    if(stepName.equals("transportRequestDocIDFromGit")) {
+                        nullScript.commonPipelineEnvironment.setValue('changeDocumentId', '300')
+                    }
+            }
+        )
+
+        def changeDocumentId = StepHelpers.getChangeDocumentId(nullScript, params)
+
+        assert nullScript.commonPipelineEnvironment.getValue('changeDocumentId') == '300'
```

Comment:
Should we not assert "changeDocumentId" here?

Reply:
done